doc/guides/west/repo-tool.rst

```diff
@@ -1,79 +1,289 @@
-in the workspace as the manifest repository. The only requirement is that it
-contains a valid manifest file. See :ref:`west-manifests` for more details on
-what this means.
-
-Both of the :file:`tinycbor` and :file:`net-tools` directories are *projects*
-managed by west, and configured in the manifest file. A west workspace can
-contain arbitrarily many projects. As shown above, projects can be located
-anywhere in the workspace. They don't have to be subdirectories of the
-manifest directory, and they can be inside of arbitrary subdirectories inside
-the workspace's root directory. By default, the Zephyr build system uses
-west to get the locations of all the projects in the workspace, so any code
-they contain can be used by applications. This behavior can be overridden using
-the ``ZEPHYR_MODULES`` CMake variable; see :ref:`modules` for details.
-
-Finally, any repository managed by a west workspace can contain
-:ref:`extension commands <west-extensions>`, which are extra west commands
-provided by that project. This includes the manifest repository and any project
-repository.
+Git repository containing a *manifest file*. The location of the manifest
+repository is given by the :ref:`manifest.path configuration option
+<west-config-index>` in the local configuration file.
+
+For upstream Zephyr, :file:`zephyr` is the manifest repository, but you can
+configure west to use any Git repository in the workspace as the manifest
+repository. The only requirement is that it contains a valid manifest file.
+See :ref:`west-topologies` for information on other options, and
+:ref:`west-manifests` for details on the manifest file format.
+
+The manifest file is a YAML file that defines *projects*, which are the
+additional Git repositories in the workspace managed by west. The manifest
+file is named :file:`west.yml` by default; this can be overridden using the
+``manifest.file`` local configuration option.
+
+You use the :ref:`west-update` command to update the workspace's projects
+based on the contents of the manifest file. The manifest file controls
+things like where the project should be placed within the workspace, what
+URL to clone it from if it's missing, and what Git revision should be
+checked out locally.
+
+Projects can be located anywhere inside the workspace, but they may not
+"escape" it. Project repositories need not be located in subdirectories of
+the manifest repository or as immediate subdirectories of the topdir.
+However, projects must have paths inside the workspace. (You may replace a
+project's repository directory within the workspace with a symbolic link to
+elsewhere on your computer, but west will not do this for you.)
+
+A workspace can contain additional Git repositories or other files and
+directories not managed by west. West basically ignores anything in the
+workspace except :file:`.west`, the manifest repository, and the projects
+specified in the manifest file.
```

Comment:
> On the other hand I don't see anything about nesting one west project inside another (assuming of course git does not mind, pretty sure it does not).

I think this is implied by:

```rst
Projects can be located anywhere inside the workspace, but they may not
"escape" it. Project repositories need not be located in subdirectories of
the manifest repository or as immediate subdirectories of the topdir.
However, projects must have paths inside the workspace.
```

Nested projects obey these rules.

Reply:
It is indeed implied by this but I think it should be explicit because nesting projects is not "obviously OK", far from it IMHO. See most recent and "scary" exemple at https://github.com/zephyrproject-rtos/west/pull/459#issuecomment-758221158

Adding "including nested projects" somewhere in the middle of that paragraph would be enough. That's just 3 words.